client/my-sites/stats/stats-list/action-follow.jsx

```diff
@@ -2,61 +2,84 @@
  * External dependencies
  */
 import React from 'react';
-import createReactClass from 'create-react-class';
-import { localize } from 'i18n-calypso';
 import classNames from 'classnames';
-import debugFactory from 'debug';
+import config from '@automattic/calypso-config';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-/* eslint-disable no-restricted-imports */
-import observe from 'calypso/lib/mixins/data-observe';
-/* eslint-enable no-restricted-imports */
-import { gaRecordEvent } from 'calypso/lib/analytics/ga';
 import Gridicon from 'calypso/components/gridicon';
+import wpcom from 'calypso/lib/wp';
+import { gaRecordEvent } from 'calypso/lib/analytics/ga';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getSiteDomain } from 'calypso/state/sites/selectors';
+import { requestSiteStats } from 'calypso/state/stats/lists/actions';
 
-const debug = debugFactory( 'calypso:stats:action-follow' );
-
-const StatsActionFollow = createReactClass( {
-	displayName: 'StatsActionFollow',
+class StatsActionFollow extends React.Component {
+	state = {
+		isFollowing: this.props.isFollowing,
+	};
 
-	mixins: [ observe( 'followSite' ) ],
+	refreshFollowers = () => {
+		this.props.requestSiteStats( this.props.selectedSiteId, 'statsFollowers', {
+			type: 'wpcom',
+			max: 10,
+		} );
+		this.props.requestSiteStats( this.props.selectedSiteId, 'statsCommentFollowers', { max: 7 } );
+	};
 
-	clickHandler: function ( event ) {
-		const site = this.props.followSite;
+	clickHandler = ( event ) => {
 		let gaEvent;
 
 		event.stopPropagation();
 		event.preventDefault();
-		debug( 'handling follower click', this.props.data );
 
-		if ( ! site.is_following ) {
+		if ( ! this.state.isFollowing ) {
 			gaEvent = 'Follow';
-			site.follow();
+			wpcom
+				.site( this.props.siteId )
+				.follow()
+				.add( { source: config( 'readerFollowingSource' ) } )
+				.then( this.refreshFollowers );
```

Comment:
What happens if the request fails? Should we add a catch and reset `isFollowing` in that case?

Reply:
Good call - it's good to not be that naive. Addressed in https://github.com/Automattic/wp-calypso/pull/49825/commits/3febfb6062febb95746955e935d4e21bb7f1de2b.